lib/storage/in-memory.js

```diff
@@ -0,0 +1,77 @@
+"use strict";
+
+/**
+ * Simulates a storage engine.  Uses process memory only.
+ * Great for testing.  Really bad for production.
+ *
+ * All methods are asynchronous.
+ */
+
+module.exports = (logger, promise) => {
+    var store;
+
+    /**
+     * Deletes a single file from configured bucket.
+     *
+     * @param {string} filename
+     * @return {Promise.<*>} promise
+     */
+    function deleteAsync(filename) {
+        if (store[filename]) {
+            logger.debug(`Delete: ${filename}`);
+            delete store[filename];
+
+            return promise.resolve();
+        }
+
+        return promise.reject(new Error(`No such key: ${filename}`));
+    }
+
+
+    /**
+     * Gets a file from S3 from the configured bucket.
+     *
+     * @param {string} filename
+     * @return {Promise.<Buffer>} indicates success/error of operation
+     */
+    function getAsync(filename) {
+        if (store[filename]) {
+            logger.debug(`Get: ${filename}`);
+            logger.debug(store[filename].toString("hex"));
+
+            return promise.resolve(store[filename]);
+        }
+
+        return promise.reject(new Error(`No such key: ${filename}`));
+    }
+
+
+    /**
+     * Uploads a file to the S3 bucket configured.
```

Comment:
Should this say "S3"?

There are other places "S3" is used in comments, should this be that way?